presto-docs/src/main/sphinx/functions/aggregate.rst

```diff
@@ -105,6 +105,41 @@ General Aggregate Functions
 
     Returns ``n`` smallest values of all input values of ``x``.
 
+.. function:: reduce_agg(inputValue T, initialState S, inputFunction(S, T, S), combineFunction(S, S, S)) -> S
+
+    Returns a single value reduced from input values. ``inputFunction`` and ``combineFunction`` will be invoked.
+    ``inputFunction`` takes the current state (initially ``initialState``) and the input value, and returns the new state.
+    ``combineFunction`` takes two states and returns the new state.
+    The value of final state is returned::
+
+        SELECT id, reduce_agg(value, 0, (a, b) -> a + b, (a, b) -> a + b)
+        FROM (
+            VALUES
+                (1, 3),
+                (1, 4),
+                (2, 5),
+                (2, 6)
+        ) AS t(id, value)
+        GROUP BY id;
+        -- (1, 7)
+        -- (2, 11)
+
+        SELECT id, reduce_agg(value, 1, (a, b) -> a * b, (a, b) -> a * b)
+        FROM (
+            VALUES
+                (1, 3),
+                (1, 4),
+                (2, 5),
+                (2, 6)
+        ) AS t(id, value)
+        GROUP BY id;
+        -- (1, 12)
+        -- (2, 30)
+
+    ``reduce_agg`` supports state with native container type long, double or boolean.
```

Comment:
I don't think any end user will understand what this means.  Maybe state this as `reduce_agg currently does not support variable width types, container types, or inexact types`.

Reply:
@dain : Maybe I will just say `reduce_agg` currently supports boolean, integer (TINYINT, SMALLINT, INTEGER, BIGINT) and floating-point (REAL, DOUBLE) types .